lib/cocoapods-core/cdn_source.rb

```diff
@@ -220,12 +233,37 @@ def git?
 
     private
 
-    # Index files contain all the sub directories in the directory, separated by
-    # a newline. We use those because you can't get a directory listing from a CDN.
-    INDEX_FILE_NAME = 'index.txt'.freeze
+    def ensure_versions_file_loaded(fragment)
+      return if !@version_arrays_by_fragment_by_name[fragment].nil? && !@check_existing_files_for_update
+
+      # Index file that contains all the versions for all the pods in the shard.
+      # We use those because you can't get a directory listing from a CDN.
+      index_file_name = index_file_name_for_fragment(fragment)
+      download_file(index_file_name)
+      versions_raw = local_file(index_file_name, &:to_a).map(&:chomp)
+      @version_arrays_by_fragment_by_name[fragment] = versions_raw.reduce({}) do |hash, row|
+        row = row.split('/')
+        pod = row.shift
+        versions = row
+
+        hash[pod] = versions
+        hash
+      end
+    end
+
+    def index_file_name_for_fragment(fragment)
+      fragment_joined = fragment.join('_')
+      fragment_joined = '_' + fragment_joined unless fragment.empty?
+      "all_pods_versions#{fragment_joined}.txt"
+    end
+
+    def pod_shard_fragment(pod_name)
```

Comment:
would love unit tests for this and `index_file_name_for_fragment`

Reply:
After trying to add tests to these I've realized that it's not a good idea:
1. These are private methods that are implementation detail.
2. This functionality is addressed in the other tests, as all the CDN calls are accounted for by Mocha, including `"all_pods_versions#{fragment_joined}.txt"`.